src/pkg/packager/deploy.go

```diff
@@ -180,7 +187,12 @@ func (p *Packager) deployComponents(ctx context.Context) ([]types.DeployedCompon
 
 		deployedComponents = append(deployedComponents, deployedComponent)
 		idx := len(deployedComponents) - 1
-
+		if p.isConnectedToCluster() {
+			if _, err := p.cluster.RecordPackageDeployment(ctx, p.cfg.Pkg, deployedComponents); err != nil {
```

Comment:
You'll need to send `packageGeneration` into this function to set `DeployedPackage.Generation`. Otherwise on subsequent deploys the package will always start at 1.